docs/usage/deriving-data-selectors.md

```diff
@@ -0,0 +1,624 @@
+
+This has several benefits:
+
+- The actual state is easier to read
+- Less logic is needed to calculate those additional values and keep them in sync with the rest of the data
+- The original state is still there as a reference and isn't being replaced
+
+:::tip
+
+This is _also_ a good principle for React state as well! Many times users tried to define a `useEffect` hook that waits for a state value to change, and then sets state with some derived value like `setAllCompleted(allCompleted)`. Instead, that value can be derived during the rendering process and used directly, without having to save the value into state at all:
+
+```js
+function TodoList() {
+  const [todos, setTodos] = useState([])
+
+  // highlight-start
+  // Derive the data while rendering
+  const allTodosCompleted = todos.every(todo => todo.completed)
+  // highlight-end
+
+  // render with this value
+}
+```
+
+:::
+
+## Calculating Derived Data with Selectors
+
+In a typical Redux application, the logic for deriving data is usually written as functions we call **_selectors_**.
+
+Selectors are primarily used to encapsulate logic for looking up specific values from state, logic for actually deriving values, and improving performance by avoiding unnecessary recalculations.
+
+You are not _required_ to use selectors for all state lookups, but they are a standard pattern and widely used.
+
+### Basic Selector Concepts
+
+**A "selector function" is any function that accepts the Redux store state (or part of the state) as an argument, and returns data that is based on that state.**
+
+Selectors don't have to be written using a special library, and it doesn't matter whether you write them as arrow functions or the `function` keyword. For example, all of these are valid selector functions:
+
+```js
+// Arrow function, direct lookup
+const selectEntities = state => state.entities
+
+// Function declaration, mapping over an array to derive values
+function selectItemIds(state) {
+  return state.items.map(item => item.id)
+}
+
+// Function declaration, encapsulating a deep lookup
+function selectSomeSpecificField(state) {
+  return state.some.deeply.nested.field
+}
+
+// Arrow function, deriving values from an array
+const selectItemsWhoseNamesStartWith = (items, namePrefix) =>
+  items.filter(item => item.name.startsWith(namePrefix))
+```
+
+A selector function can have any name you want. However, [**we recommend prefixing selector function names with the word `select` combiend with a description of the value being selected**](../style-guide/style-guide.md#name-selector-functions-as-selectthing). Typical examples of this would look like **`selectTodoById`**, **`selectFilteredTodos`**, and **`selectVisibleTodos**`.
```

Comment:
```suggestion
A selector function can have any name you want. However, [**we recommend prefixing selector function names with the word `select` combined with a description of the value being selected**](../style-guide/style-guide.md#name-selector-functions-as-selectthing). Typical examples of this would look like **`selectTodoById`**, **`selectFilteredTodos`**, and **`selectVisibleTodos**`.
```